packages/discord.js/src/structures/BaseGuildVoiceChannel.js

```diff
@@ -3,12 +3,30 @@
 const { Collection } = require('@discordjs/collection');
 const { PermissionFlagsBits } = require('discord-api-types/v10');
 const GuildChannel = require('./GuildChannel');
+const TextBasedChannel = require('./interfaces/TextBasedChannel');
+const MessageManager = require('../managers/MessageManager');
 
 /**
  * Represents a voice-based guild channel on Discord.
  * @extends {GuildChannel}
+ * @implements {TextBasedChannel}
  */
 class BaseGuildVoiceChannel extends GuildChannel {
+  constructor(guild, data, client) {
+    super(guild, data, client, false);
+    /**
+     * A manager of the messages sent to this channel
+     * @type {MessageManager}
+     */
+    this.messages = new MessageManager(this);
+
+    /**
+     * If the guild considers this channel NSFW
+     * @type {boolean}
+     */
+    this.nsfw = Boolean(data.nsfw);
+  }
```

Comment:
See my other comment
```suggestion
    this.nsfw = Boolean(data.nsfw);

    this._patch(data);
  }
```